### PR TITLE
TECH-307 - Basic public api

### PR DESCRIPTION
**Pull request by Gcaya**
Adding the basic public api pieces without any logic in them.

### CODE REVIEW

**Review by Zav — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -1,36 +1,67 @@
-# GSoft.Authentication.Client.Credentials.Grant
+# GSoft.Authentication.ClientCredentialsGrant
 
 [![nuget](https://img.shields.io/nuget/v/GSoft.Authentication.Client.Credentials.Grant.svg?logo=nuget)](https://www.nuget.org/packages/GSoft.Authentication.Client.Credentials.Grant/)
```
Comment (asimmon) [inline README.md:3]:
```suggestion
[![nuget](https://img.shields.io/nuget/v/GSoft.Extensions.Http.Authentication.ClientCredentialsGrant.svg?logo=nuget)](https://www.nuget.org/packages/GSoft.Extensions.Http.Authentication.ClientCredentialsGrant/)
```

File: README.md [hunk 2]
```diff
@@ -1,36 +1,67 @@
...
 [![build](https://img.shields.io/github/actions/workflow/status/gsoft-inc/gsoft-authentication-client-credentials-grant/publish.yml?logo=github&branch=main)](https://github.com/gsoft-inc/gsoft-authentication-client-credentials-grant/actions/workflows/publish.yml)
```
Comment (asimmon) [inline README.md:4]: I wonder if we should rename the github repo
* from `gsoft-authentication-client-credentials-grant`
* to `gsoft-authentication-clientcredentialsgrant` ?

If we want to do that, use VS Code to ensure we find & replace any reference

File: README.md [hunk 3]
```diff
@@ -1,36 +1,67 @@
...
-* Class library targeting .NET Standard, a Xunit-based test project, both built with .NET 6 SDK.
-* Public API breaking changes detection using [Microsoft.CodeAnalysis.PublicApiAnalyzers](https://github.com/dotnet/roslyn-analyzers/blob/main/src/PublicApiAnalyzers/PublicApiAnalyzers.Help.md), which is also used in .NET and many other open-souce projects source code.
-* Shared csproj properties with a `Directory.Build.props` file.
-* A `Build.ps1` script that can be executed to simulate a CI build locally.
-* A public `*.snk` key used to sign the assembly and make it [strong-named](https://learn.microsoft.com/en-us/dotnet/standard/assembly/strong-named). This is required by some ShareGate projects. It is also an industry standard: Microsoft, Polly, MediatR, Newtonsoft.Json, Moq, FluentAssertions, etc.
-* [Source Link](https://github.com/dotnet/sourcelink) support.
-* Issue templates, Apache License 2.0, markdown files for contributing and security required by GSec.
-
+This library offers an IHttpClientBuilder extension method for streamlined access token retrieval and caching using the OAuth 2.0 Client Credentials Grant flow.
```
Comment (asimmon) [inline README.md:6]:
```suggestion
This library offers an `IHttpClientBuilder` extension method for streamlined access token retrieval and caching using the OAuth 2.0 Client Credentials Grant flow.
```

File: README.md [hunk 4]
```diff
@@ -1,36 +1,67 @@
...
+      "Scopes": [
+        "scope1",
+        "scope2"
+      ]
+    },
+    "Service2": {
+      "Authority": "<authority_url>",
+      "ClientId": "<client_id>",
+      "ClientSecret": "<client_secret>",
+      "Scopes": [
```
Comment (asimmon) [inline README.md:32]: Maybe we want to show the singular `Scope` property?

File: README.md [hunk 5]
```diff
@@ -1,36 +1,67 @@
...
-
+Install the package `GSoft.Authentication.ClientCredentialsGrant` in the project where you want to register an HttpClient.
+This package contains the extension method that adds the access token management to an HttpClient.
+By default the library will bind options based on the name used to register the HttpClient, this allows having more than one client credentials configuration.
+
+## Example
+```csharp
+// appsettings.json
+{
+  "ClientCredentialsHttpClients": {
```
[the remark this answers is not in the document]
Comment (asimmon) [reply README.md:18]: @Zav Let's talk privately about this!

File: src/GSoft.Extensions.Http.Authentication.ClientCredentialsGrant/ClientCredentialsOptions.cs [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    /// Confidential secret generated by the Authority and shared with the client application.
+    /// </summary>
+    public string ClientSecret { get; set; } = string.Empty;
+
+    /// <summary>
+    /// Represents the permissions or access levels requested by the client application from the Authority.
+    /// </summary>
+    public string[] Scopes { get; set; } = Array.Empty<string>();
+
+    public string Scope
```
Comment (asimmon) [inline src/GSoft.Extensions.Http.Authentication.ClientCredentialsGrant/ClientCredentialsOptions.cs:32]: Missing documentation here

File: README.md [hunk 7]
```diff
@@ -1,36 +1,67 @@
...
 ## Getting started
 
-TODO
-
+Install the package `GSoft.Authentication.ClientCredentialsGrant` in the project where you want to register an HttpClient.
+This package contains the extension method that adds the access token management to an HttpClient.
+By default the library will bind options based on the name used to register the HttpClient, this allows having more than one client credentials configuration.
+
+## Example
+```csharp
```
[the remark this answers is not in the document]
Comment (Gcaya) [reply README.md:15]: Good point!

File: README.md (hunk 2 above)
Comment (Gcaya) [reply README.md]: It's done!

**Review by asimmon — APPROVED** [review]

**Pull request merged**